src/fusedav.c

```diff
@@ -188,25 +188,52 @@ static void getdir_propfind_callback(__unused void *userdata, const char *path,
     memset(&value, 0, sizeof(struct stat_cache_value));
     value.st = st;
 
-    log_print(LOG_INFO, SECTION_FUSEDAV_PROP, "%s: %s (%lu)", funcname, path, status_code);
+    log_print(LOG_INFO, SECTION_FUSEDAV_PROP, "%s: %s (%lu)", 
+            funcname, path, status_code);
 
     if (status_code == 410) {
         struct stat_cache_value *existing;
 
-        log_print(LOG_INFO, SECTION_FUSEDAV_PROP, "%s: DELETE %s (%lu)", funcname, path, status_code);
+        log_print(LOG_INFO, SECTION_FUSEDAV_PROP, "%s: DELETE %s (%lu)", 
+                funcname, path, status_code);
         existing = stat_cache_value_get(config->cache, path, true, &subgerr1);
         if (subgerr1) {
             g_propagate_prefixed_error(gerr, subgerr1, "%s: ", funcname);
             return;
         }
 
+        /* Binding A does a propfind.
+         * Binding B does a propfind on a path with multiple directories.
+         * Say, /abcdir. Binding B starts the propfind at /, and abcdir
+         * still exists. Then Binding A deletes the directory abcdir.
+         * Then Binding B's propfind moves to abcdir where it does the
+         * next stage in its propfind. But abcdir no longer exists. 
+         * Valhalla returns 404, which gets processed and this function
```

Comment:
This "`Valhalla returns 404`" might be a little confusing because we're currently inside of an `if (status_code == 410)`.  Is it worth expanding how we get from 404 to 410?